test/battle/ability/shield_dust.c

```diff
@@ -0,0 +1,124 @@
+        ASSUME(MoveHasMoveEffectWithChance(MOVE_THOUSAND_ARROWS, MOVE_EFFECT_SMACK_DOWN, 0) == TRUE);
+        ASSUME(MoveHasMoveEffectWithChance(MOVE_JAW_LOCK, MOVE_EFFECT_TRAP_BOTH, 0) == TRUE);
+        ASSUME(MoveHasMoveEffectWithChance(MOVE_PAY_DAY, MOVE_EFFECT_PAYDAY, 0) == TRUE);
+        ASSUME(MoveHasMoveEffectWithChance(MOVE_SMACK_DOWN, MOVE_EFFECT_SMACK_DOWN, 0) == TRUE);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_VIVILLON) { Ability(ABILITY_SHIELD_DUST); }
+    } WHEN {
+        TURN { MOVE(player, move); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, move, player);
+        HP_BAR(opponent);
+        switch (move)
+        {
+            case MOVE_INFESTATION:
+                MESSAGE("Foe Vivillon has been afflicted with an infestation by Wobbuffet!");
+                break;
+            case MOVE_THOUSAND_ARROWS:
+                MESSAGE("Foe Vivillon fell straight down!");
+                break;
+            case MOVE_JAW_LOCK:
+                MESSAGE("Neither Pokémon can run away!");
+                break;
+            case MOVE_PAY_DAY:
+                MESSAGE("Coins scattered everywhere!");
+                break;
+        }
+    } THEN { // Can't find good way to test trapping
+        if (move == MOVE_JAW_LOCK) {
+            EXPECT(opponent->status2 & STATUS2_ESCAPE_PREVENTION);
+            EXPECT(player->status2 & STATUS2_ESCAPE_PREVENTION);
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Shield Dust does not block self-targeting effects, primary or secondary")
+{
+    u16 move;
+    PARAMETRIZE { move = MOVE_POWER_UP_PUNCH; }
+    PARAMETRIZE { move = MOVE_RAPID_SPIN; }
+    PARAMETRIZE { move = MOVE_LEAF_STORM; }
+    PARAMETRIZE { move = MOVE_METEOR_ASSAULT; }
+
+    GIVEN {
+        ASSUME(MoveHasMoveEffectSelf(MOVE_POWER_UP_PUNCH, MOVE_EFFECT_ATK_PLUS_1) == TRUE);
+        ASSUME(MoveHasMoveEffectSelf(MOVE_RAPID_SPIN, MOVE_EFFECT_RAPIDSPIN) == TRUE);
+        ASSUME(MoveHasMoveEffectSelf(MOVE_LEAF_STORM, MOVE_EFFECT_SP_ATK_TWO_DOWN) == TRUE);
+        ASSUME(MoveHasMoveEffectSelf(MOVE_METEOR_ASSAULT, MOVE_EFFECT_RECHARGE) == TRUE);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_VIVILLON) { Ability(ABILITY_SHIELD_DUST); }
+    } WHEN {
+        TURN { MOVE(player, move); }
+        if (move == MOVE_METEOR_ASSAULT) {
+            TURN { SKIP_TURN(player); }
+        }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, move, player);
+        HP_BAR(opponent);
+        switch (move)
+        {
+            case MOVE_POWER_UP_PUNCH:
+            case MOVE_RAPID_SPIN:
+            case MOVE_LEAF_STORM:
+                ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
+                break;
+            case MOVE_METEOR_ASSAULT: // second turn
+                MESSAGE("Wobbuffet must recharge!");
+                break;
+        }
+    }
+}
+
+TO_DO_BATTLE_TEST("Shield Dust does not block Sparkling Aria in doubles"); // Would certainly fail at the moment
```

Comment:
Let's create a known failing for this

Reply:
Oof, I don't really want to as part of this, tbh - this is just a bugfix PR for #4104 . We can open an issue for it, but it's such a minor mechanic...

Reply:
Let me write one fast so we don't waste an issue

Reply:
> Let me write one fast so we don't waste an issue

Appreciate it :)

Reply:
Isn't this an easy fix, or is there a problem with excluding Sparkling Aria effect?
```
DOUBLE_BATTLE_TEST("Shield Dust does not block Sparkling Aria in doubles")
{
    KNOWN_FAILING;
    GIVEN {
        PLAYER(SPECIES_WOBBUFFET);
        PLAYER(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_VIVILLON) { Ability(ABILITY_SHIELD_DUST); Status1(STATUS1_BURN); }
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(playerLeft, MOVE_SPARKLING_ARIA); }
    } SCENE {
        ANIMATION(ANIM_TYPE_MOVE, MOVE_SPARKLING_ARIA, playerLeft);
        NOT MESSAGE("Foe Vivillion's burn was healed.");
        STATUS_ICON(opponentLeft, burn: TRUE);
    }
}
```

Reply:
Then add the test to the PR so it can be merged 😅